### PR TITLE
Database: sqlite to postgres

### PR DESCRIPTION
**Pull request by vishnukvnvv**

### CODE REVIEW

File: bundler/config/Staging.toml [hunk 1]
```diff
@@ -15,6 +15,7 @@ level = "info"
 
 [database]
 file = "toad.db"
+url = "postgres://<user>:<pwd>@<host>/<db_name>"
```
Comment (ska1296) [inline bundler/config/Staging.toml:18]: I think you can fill in everything except password here

File: bundler/src/db/dao/wallet_dao.rs [hunk 2]
```diff
@@ -1,54 +1,40 @@
...
-
-        if !rows.is_empty() {
-            return rows[0].wallet_address.to_string();
-        }
-        return "".to_string();
+        let query = query_as!(User, "SELECT * from users where email = $1", user_id);
+        let result: Result<User, Error> = query.fetch_one(&self.pool).await;
+        return match result {
+            Ok(user) => user.wallet_address,
+            Err(_) => "".to_string(),
```
Comment (ska1296) [inline bundler/src/db/dao/wallet_dao.rs:27]: log the error

File: bundler/src/db/dao/wallet_dao.rs [hunk 3]
```diff
@@ -1,54 +1,40 @@
...
-                salt: rows[0].salt.clone(),
-                deployed: rows[0].deployed,
-            });
-        }
-        return None;
+        let query = query_as!(User, "SELECT * from users where email = $1", user_id);
+        let result: Result<Option<User>, Error> = query.fetch_optional(&self.pool).await;
+        return match result {
+            Ok(user) => user,
+            Err(_) => None,
```
Comment (ska1296) [inline bundler/src/db/dao/wallet_dao.rs:36]: log the error

File: bundler/src/db/dao/wallet_dao.rs [hunk 4]
```diff
@@ -58,36 +44,17 @@ impl WalletDao {
...
+        let query = query!(
+            "INSERT INTO users (email, wallet_address, salt, deployed) VALUES ($1, $2, $3, $4)",
+            user_id,
+            wallet_address,
+            salt,
+            deployed
+        );
+        let result = query.execute(&self.pool).await;
+        if result.is_err() {
+            warn!("Failed to create user: {}", user_id);
```
Comment (ska1296) [inline bundler/src/db/dao/wallet_dao.rs:56]: log the error too

File: bundler/src/models/transaction/list_transactions_params.rs [hunk 5]
```diff
@@ -3,5 +3,5 @@ use serde::Deserialize;
 #[derive(Deserialize)]
 pub struct ListTransactionsParams {
     pub id: Option<i32>,
-    pub page_size: Option<i32>,
+    pub page_size: Option<i64>,
```
Comment (ska1296) [inline bundler/src/models/transaction/list_transactions_params.rs:6]: why can't page size continue to be i32?

File: bundler/src/models/config/settings.rs [hunk 6]
```diff
@@ -28,6 +28,7 @@ pub struct Server {
 #[derive(Debug, Deserialize, Clone)]
 pub struct Database {
     pub file: String,
+    pub url: String,
```
Comment (ska1296) [inline bundler/src/models/config/settings.rs:31]: add a function `get_url()` that reads the user and password from env variable and puts it in URL the string. The db and host can be hardcoded.

File: bundler/src/server.rs [hunk 7]
```diff
@@ -79,10 +79,10 @@ pub fn init_services() -> ToadService {
     );
 
     //daos
-    let pool = establish_connection(CONFIG.database.file.clone());
+    let pool = DatabaseConnection::init(CONFIG.database.url.clone()).await;
```
Comment (ska1296) [inline bundler/src/server.rs:82]: replace this with "get_url()"

**Review by ska1296 — CHANGES_REQUESTED** [review]

File: bundler/src/db/connection.rs [hunk 8]
```diff
@@ -1,7 +1,20 @@
...
 
-pub fn establish_connection(database_url: String) -> Pool<SqliteConnectionManager> {
-    let manager = SqliteConnectionManager::file(&database_url);
-    Pool::new(manager).unwrap()
+pub struct DatabaseConnection {}
+
+impl DatabaseConnection {
+    pub async fn init() -> sqlx::Pool<Postgres> {
+        let database_url: String;
+        match std::env::var("DATABASE_URL") {
```
Comment (ska1296) [inline bundler/src/db/connection.rs:10]: update .env.example with a sample url

File: bundler/src/db/dao/metadata_dao.rs [hunk 9]
```diff
@@ -15,58 +12,55 @@ impl MetadataDao {
...
+        };
     }
 }
 
 #[derive(Default, Clone)]
 pub struct SupportedCurrency {
     pub chain: String,
     pub currency: String,
-    pub exponent: u8,
+    pub exponent: i32,
```
Comment (defyclub) [inline bundler/src/db/dao/metadata_dao.rs:72]: Don't think this matters much, but we we require a i32 here?

File: bundler/src/db/dao/wallet_dao.rs [hunk 10]
```diff
@@ -58,36 +54,21 @@ impl WalletDao {
...
+        let query = query!(
+            "INSERT INTO users (email, wallet_address, salt, deployed) VALUES ($1, $2, $3, $4)",
+            user_id,
+            wallet_address,
+            salt,
+            deployed
+        );
+        let result = query.execute(&self.pool).await;
+        if result.is_err() {
+            warn!(
```
Comment (ska1296) [inline bundler/src/db/dao/wallet_dao.rs:66]:
```suggestion
            error!(
```

File: bundler/src/db/dao/wallet_dao.rs [hunk 11]
```diff
@@ -1,54 +1,50 @@
...
-            .unwrap();
-        stmt.execute([true.to_string(), user_id]).unwrap();
+        let query = query!(
+            "UPDATE users SET deployed = $1 WHERE email = $2",
+            true,
+            user_id
+        );
+        let result = query.execute(&self.pool).await;
+        if result.is_err() {
+            warn!(
```
Comment (ska1296) [inline bundler/src/db/dao/wallet_dao.rs:18]: error!

**Review by ska1296 — CHANGES_REQUESTED** [review]

**Review by ska1296 — APPROVED** [review]

**Pull request merged**